File: projects/punk0/code/sort.py
# sort the list in ascending order
#   [1,4,2,3]
# returns:
#   [1,2,3,4]
def sort(x):
    # we can use the sorted() function
    # make a sorted copy

    # x.sort method mutates the list so
    # first we need to copy it and then
    # sort the copy
    r = []
    for v in x:
        r.append(v)

    r.sort()

    return r


print(sort([1, 1, 2, 3, 3, 4, 1, 2, 7, 9]))

File: projects/punk0/code/sort.js
// sort the list in ascending order
//   [5,1,4,2,3]
// returns:
//   [1,2,3,4,5]
function sort(x) {
  let r = [];
  for (let v of x) {
    r.push(v);
  }

  r.sort();

  return r;
}

console.log(JSON.stringify(sort([1, 1, 2, 3, 3, 4, 1, 2, 7, 9])));
